src/main/java/fr/sncf/osrd/infra_state/SwitchState.java

```diff
@@ -1,108 +1,106 @@
 package fr.sncf.osrd.infra_state;
 
-import static fr.sncf.osrd.infra.trackgraph.SwitchPosition.LEFT;
-import static fr.sncf.osrd.infra.trackgraph.SwitchPosition.MOVING;
-
-import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
-import fr.sncf.osrd.infra.railscript.value.RSMatchable;
-import fr.sncf.osrd.infra.railscript.value.RSValue;
 import fr.sncf.osrd.infra.trackgraph.Switch;
-import fr.sncf.osrd.infra.trackgraph.SwitchPosition;
 import fr.sncf.osrd.infra.trackgraph.TrackSection;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import fr.sncf.osrd.infra.railscript.value.RSValue;
 import fr.sncf.osrd.infra_state.events.SwitchMoveEvent;
 import fr.sncf.osrd.simulation.EntityChange;
 import fr.sncf.osrd.simulation.Simulation;
 import fr.sncf.osrd.simulation.SimulationError;
+import fr.sncf.osrd.utils.graph.EdgeDirection;
+import fr.sncf.osrd.utils.graph.EdgeEndpoint;
 
 /**
  * The state of the route is the actual entity which interacts with the rest of the infrastructure
  */
-public final class SwitchState implements RSMatchable {
+public final class SwitchState implements RSValue {
     public final Switch switchRef;
 
-    private SwitchPosition position;
+    private String group;
 
     public SwitchState(Switch switchRef) {
         this.switchRef = switchRef;
-        this.position = LEFT;
+        this.group = switchRef.getDefaultGroup();
     }
 
-    public SwitchPosition getPosition() {
-        return position;
+    public String getGroup() {
+        return group;
     }
 
     /**
      * Return currently active branch
      */
-    public TrackSection getBranch() {
-        switch (position) {
-            case LEFT:
-                return switchRef.leftTrackSection;
-            case RIGHT:
-                return switchRef.rightTrackSection;
-            default:
-                return null;
+    public TrackSection getBranch(TrackSection trackSection, EdgeDirection direction) {
+        var endpoint = direction == EdgeDirection.START_TO_STOP
+                ? EdgeEndpoint.END
+                : EdgeEndpoint.BEGIN;
+        var edges = switchRef.groups.get(group);
```

Comment:
This may be null if there is an error in the infra and the group doesn't match anything. Maybe we should validate that the position is correct during the infra/route initialization and throw an InvalidInfraException otherwise, to avoid obscure 500s later on.